ts/Studio.ts

```diff
@@ -1131,6 +1143,83 @@ export class Studio {
             // Nothing to see
             return;
         }
+
+        const addStat = (text: string, input: Node, table: HTMLTableElement, title?: string, rowClass?: string) => {
+            const tr = document.createElement('tr');
+            if (rowClass) {
+                tr.classList.add(rowClass);
+            }
+            let td = document.createElement('td');
+            td.classList.add('col-2', 'text-nowrap');
+            td.textContent = text;
+            if (title) td.title = title;
+            tr.appendChild(td);
+            td = document.createElement('td');
+            td.classList.add('col-auto');
+            td.appendChild(input);
+            tr.appendChild(td);
+            table.append(tr);
+        };
+
+        if (editAll) {
+            const editFrame = <Frame>{
+                location: <Vector>{
+                    x: 0,
+                    y: 0,
+                    z: 0,
+                },
+                rotation: <Rotator>{
+                    pitch: 0,
+                    yaw: 0,
+                    roll: 0,
+                },
+            };
+
+            tr = document.createElement('tr');
+            tbody.appendChild(tr);
+            // Type
+            let td = document.createElement('td');
+            td.textContent = 'EDIT ALL';
+            tr.appendChild(td);
+            // Name
+            td = document.createElement('td');
+            td.textContent = 'N/A';
+            tr.appendChild(td);
+            // Number
+            td = document.createElement('td');
+            td.textContent = 'N/A';
+            tr.appendChild(td);
+            // State table
+            td = document.createElement('td');
+            tr.appendChild(td);
+            const table2 = document.createElement('table');
+            table2.classList.add('table', 'table-borderless', 'mb-0');
+            td.appendChild(table2);
+            const tbody2 = document.createElement('tbody');
+            table2.appendChild(tbody2);
+
+            // Location
+            const setFrameLocation = (locationOffset: Vector) => {
+                filtered.forEach((frame: Frame) => {
+                    frame.location = vectorSum(frame.location, locationOffset);
+                });
+
+                return editFrame.location = locationOffset;
+            };
+            addStat('Location', editVector(this, editFrame.location, setFrameLocation), table2);
+            // Rotation
+            const setFrameRotation = (rotationOffset: Rotator) => {
+                filtered.forEach((frame: Frame) => {
+                    frame.rotation.pitch = (frame.rotation.pitch + rotationOffset.pitch) % 180;
+                    frame.rotation.roll = (frame.rotation.roll + rotationOffset.roll) % 180;
+                    frame.rotation.yaw = (frame.rotation.yaw + rotationOffset.yaw) % 180;
```

Comment:
I'm not sure that this rotation tool makes sense - what is the situation where you would want to add a specific value to the Euler angles of all rolling stock?

The math also looks suspicious. Did you mean to use mod 360? There's a utility function called `normalizeAngle` that will wrap an input angle to the range `(-180..180]` which may be more appropriate here, as in:

```suggestion
                    frame.rotation.pitch = normalizeAngle(frame.rotation.pitch + rotationOffset.pitch);
                    frame.rotation.roll = normalizeAngle(frame.rotation.roll + rotationOffset.roll);
                    frame.rotation.yaw = normalizeAngle(frame.rotation.yaw + rotationOffset.yaw);
```

Reply:
Mod 180 works if the data is actually stored as -180 to 180. Mod handles negatives.